From cb9796bd48ef30fba6383bfbab99d4722a5f0c76 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Fri, 24 Jan 2025 06:18:38 +0100
Subject: [PATCH] Load libz3.so via ctypes.util.find_library

In contrast to the current code, this allows us to load versioned
.so files. In Alpine, this has the consequence that we don't have
to depend on the -dev subpackage.

See:

* https://gitlab.alpinelinux.org/alpine/aports/-/issues/16856
* https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/78935

---
 scripts/update_api.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)
 mode change 100755 => 100644 scripts/update_api.py

diff --git a/scripts/update_api.py b/scripts/update_api.py
old mode 100755
new mode 100644
index 153052deb..f41016c27
--- a/scripts/update_api.py
+++ b/scripts/update_api.py
@@ -1833,6 +1833,7 @@ import atexit
 import sys, os
 import contextlib
 import ctypes
+import ctypes.util
 if sys.version_info >= (3, 9):
     import importlib.resources as importlib_resources
 else:
@@ -1888,7 +1889,10 @@ for d in _all_dirs:
 if _lib is None:
   # If all else failed, ask the system to find it.
   try:
-    _lib = ctypes.CDLL('libz3.%s' % _ext)
+    name = ctypes.util.find_library("z3")
+    if name is None:
+      raise Z3Exception('cannot determine path for libz3.%s' % _ext)
+    _lib = ctypes.CDLL(name)
   except Exception as e:
     _failures += [e]
     pass
